pandas/core/tools/datetimes.py

```diff
@@ -89,6 +93,7 @@
 DatetimeScalar = Union[Scalar, datetime]
 
 DatetimeScalarOrArrayConvertible = Union[DatetimeScalar, ArrayConvertible]
+DictConvertible = Union[Dict[str, List], "DataFrame"]
```

Comment:
looks like the relevant check is for abc.MutableMapping.  any reason not to use that?

Reply:
It's too broad.  We only accept something that could be passed to `pd.DataFrame()` as a dictionary (with specific keys "year", "month", "day" at a minimum, with values that are lists).  `abc.MutableMapping` is much wider, e.g., `{123: "abc"}` is acceptable from a typing standpoint.

Reply:
> with specific keys "year", "month", "day" at a minimum

If there are keys that always need to be present, could use a TypedDict.

Reply:
> If there are keys that always need to be present, could use a TypedDict.

Nice idea.  Did that in commit https://github.com/pandas-dev/pandas/pull/46273/commits/b451ca836dacfb0247d682564b30197c51218a32